### PR TITLE
small cleanups & dead code

### PR DESCRIPTION
**Pull request by nikodemus**
Please review one commit at a time :)

### CODE REVIEW

File: src/deluge/model/note/note.h [hunk 1]
```diff
@@ -42,10 +42,10 @@ class Note : public Positionable {
 	inline int32_t getProbability() { return probability; }
 
 	//	void writeToFile();
-
 	int32_t length;
 	uint8_t velocity;
 	uint8_t probability;
+	// Lift is noteOff velocity.
```
Comment (sapphire-arches) [inline src/deluge/model/note/note.h:48]:
```suggestion
	/// Lift is noteOff velocity.
```

File: src/deluge/util/container/array/ordered_resizeable_array.h [hunk 2]
```diff
@@ -32,9 +32,18 @@ class OrderedResizeableArray : public ResizeableArray {
...
+	// TODO: move to unit tests
 	void test();
-	void testSequentiality(char const* errorCode);
+#endif
+#if TEST_VECTOR_DUPLICATES
+	// TODO: move to unit tests
 	void testDuplicates();
+#endif
+#if ENABLE_SEQUENTIALITY_TESTS
+	// This is used in assert-like fashion, so not for unit tests.
```
Comment (sapphire-arches) [inline src/deluge/util/container/array/ordered_resizeable_array.h:44]:
```suggestion
	/// \brief test that the keys in this array are sorted in ascending order. 
	///
	/// This is used in assert-like fashion, so not for unit tests.
```

File: src/deluge/util/container/array/ordered_resizeable_array.h [hunk 3]
```diff
@@ -32,9 +32,20 @@ class OrderedResizeableArray : public ResizeableArray {
...
+#endif
+#if TEST_VECTOR_DUPLICATES
+	// TODO: move to unit tests
 	void testDuplicates();
+#endif
+#if ENABLE_SEQUENTIALITY_TESTS
+	/// \brief test that the keys in this array are sorted in ascending order.
+	///
+	/// This is used in assert-like fashion, so not for unit tests.
+	void testSequentiality(char const* errorCode);
```
Comment (m-m-adams) [inline src/deluge/util/container/array/ordered_resizeable_array.h:47]: I'm not sure this is actually better, the whole function gets optimized out in the old way and it's more readable

Comment (nikodemus) [reply src/deluge/util/container/array/ordered_resizeable_array.h:46]: reverted

**Pull request merged**